testsuite/forge/src/success_criteria.rs

```diff
@@ -252,38 +263,52 @@ impl SuccessCriteria {
     }
 }
 
-enum CombinedError {
-    Single(anyhow::Error),
-    Multiple(CriteriaCheckerErrors),
+#[derive(Clone, Debug, Eq, PartialEq)]
+enum CheckType {
+    Hard,
+    Soft,
+}
+
+#[derive(Debug)]
+struct CheckError {
+    name: String,
```

Comment:
wonder if this should instead be an enum. We could then use them directly as metric label values, which might be useful in identifying an increasing number of soft failures. Otherwise soft failures we will eventually just ignore.